### PR TITLE
BLS Signature Verification Example

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
added example contract for verifying a BLS Signature using Keccak hashing algorithm

Added test suite, and added test suite to CI. 

Also added some missing tests from the publish results.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/bls-signature/bls-signature-verification.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  before(async function () {
+    signers = await ethers.getSigners();
+    const BLSTestFactory = await ethers.getContractFactory(
+      Constants.Contract.BLSTest
+    );
+
+    BLS = await BLSTestFactory.deploy();
+
+    console.log(
```
Comment (quiet-node) [inline test/bls-signature/bls-signature-verification.js:40]: Do we need this log? I know it's for debugging purpose but probably we can have another deployment unit test and make sure that the BLS contract is deployed by expect its address to exist and not be zero address?

File: test/bls-signature/bls-signature-verification.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    );
+  });
+
+  it('should verify a valid signature', async () => {
+    let signatureX =
+      '11181692345848957662074290878138344227085597134981019040735323471731897153462';
+    let signatureY =
+      '6479746447046570360435714249272776082787932146211764251347798668447381926167';
+    let result = BLS.verify(message, signatureX, signatureY);
+    console.log('Transaction hash (valid signature):', result.tx);
```
Comment (quiet-node) [inline test/bls-signature/bls-signature-verification.js:51]: Instead of logging, we should make assertion.

File: test/bls-signature/bls-signature-verification.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  });
+
+  it('should not verify a invalid signature', async () => {
+    try {
+      let signatureX =
+        '11181692345848957662074290878138344227085597134981019040735323471731897153462';
+      let signatureY = '12345';
+      let result = await BLS.verify(message, signatureX, signatureY);
```
Comment (quiet-node) [inline test/bls-signature/bls-signature-verification.js:59]: We should add a falsy statement here to force the program to go to the `catch` block, e.g.  `expect(true).to.be.false;`. Otherwise in case this line does not throw the error, this UT will pass but not in a way that we expect

File: test/bls-signature/bls-signature-verification.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  });
+
+  it('should not verify a invalid message', async () => {
+    try {
+      let signatureX =
+        '11181692345848957662074290878138344227085597134981019040735323471731897153462';
+      let signatureY =
+        '6479746447046570360435714249272776082787932146211764251347798668447381926167';
+      const result = BLS.verify('0x12345', signatureX, signatureY);
```
Comment (quiet-node) [inline test/bls-signature/bls-signature-verification.js:76]: We should add a falsy statement here to force the program to go to the `catch` block, e.g.  `expect(true).to.be.false;`. Otherwise in case this line does not throw the error, this UT will pass.

File: test/bls-signature/bls-signature-verification.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    } catch (err) {
+      console.log(err.message);
```
Comment (quiet-node) [inline test/bls-signature/bls-signature-verification.js:78]: Do we need this log?

**Review by quiet-node — CHANGES_REQUESTED** [review]
Good work. I left some suggestions

File: contracts-abi/contracts/bls-signature/Migrations.sol/Migrations.json [hunk 6]
```diff
@@ -0,0 +1,59 @@
+[
```
Comment (quiet-node) [inline contracts-abi/contracts/bls-signature/Migrations.sol/Migrations.json:1]: you might want to remove this ABI since the contract was removed

File: contracts/bls-signature/BLS.sol [hunk 7]
```diff
@@ -0,0 +1,29 @@
+pragma solidity ^0.8.0;
```
Comment (quiet-node) [inline contracts/bls-signature/BLS.sol:1]: missing license header

File: contracts/bls-signature/Pairing.sol [hunk 8]
```diff
@@ -0,0 +1,155 @@
+pragma solidity ^0.8.0;
```
Comment (quiet-node) [inline contracts/bls-signature/Pairing.sol:1]: missing license header

**Review by quiet-node — CHANGES_REQUESTED** [review]

**Review by quiet-node — APPROVED** [review]
LG but one Q

File: contracts/bls-signature/Pairing.sol [hunk 9]
```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+/*
+ * @title Pairing
+ * @dev BN128 pairing operations.
+ * @ref https://github.com/JacobEberhardt/ZoKrates/blob/da5b13f845145cf43d555c7741158727ef0018a2/zokrates_core/src/verification.rs
+ */
+
+library Pairing {
```
[the remark this answers is not in the document]
Comment (AlfredoG87) [reply contracts/bls-signature/Pairing.sol:10]: Maybe, it would be tricky but possible.
However with BLSTest, all the methods are being indirectly covered.
